src/mmw/js/src/core/modals/views.js

```diff
@@ -75,6 +87,10 @@ var ConfirmView = ModalBaseView.extend({
     }, ModalBaseView.prototype.events),
 
     primaryAction: function() {
+        if (this.getDisabledState(this.ui.confirmation) === true) {
```

Comment:
Since you created getDisabledState, you know that it returns a boolean so you could just write
`if (this.getDisabledState(this.ui.confirmation)) { ...`